src/data/equiv/basic.lean

```diff
@@ -353,6 +353,30 @@ protected lemma image_compl {α β} (f : equiv α β) (s : set α) :
   f '' sᶜ = (f '' s)ᶜ :=
 set.image_compl_eq f.bijective
 
+@[simp] lemma symm_preimage_preimage {α β} (e : α ≃ β) (s : set β) :
+  e.symm ⁻¹' (e ⁻¹' s) = s :=
+by ext; simp
+
+@[simp] lemma preimage_symm_preimage {α β} (e : α ≃ β) (s : set α) :
+  e ⁻¹' (e.symm ⁻¹' s) = s :=
+by ext; simp
+
+@[simp] lemma preimage_subset {α β} (e : α ≃ β) (s t : set β) : e ⁻¹' s ⊆ e ⁻¹' t ↔ s ⊆ t :=
+⟨by simpa using @set.preimage_mono _ _ e.symm (e ⁻¹' s) (e ⁻¹' t),
+ set.preimage_mono⟩
+
+@[simp] lemma image_subset {α β} (e : α ≃ β) (s t : set α) : e '' s ⊆ e '' t ↔ s ⊆ t :=
+⟨by { rw ← e.preimage_subset, simp }, set.image_subset _⟩
+
+@[simp] lemma image_eq_iff_eq {α β} (e : α ≃ β) (s t : set α) : e '' s = e '' t ↔ s = t :=
```

Comment:
Unfortunately, we need these lemmas both for embeddings and equivs, because the embedding lemmas don't match the equiv coercions...

Reply:
Are the proofs at least defeq, such that you can prove this lemma with the embedding version?

Reply:
I'm not sure if we even have all of them for embeddings.  Three of them already exist with an `injective` or `surjective` assumption.